doc/.gitignore

```diff
@@ -1,6 +1,2 @@
-html
-temp
-out.txt
-qbk/reference.qbk
```

Comment:
I think we could keep these without detriment.

Reply:
I removed them to keep everything tidy since we won't be using BoostBook again.